move clip-selector to extras/

dylanjha opened the pull request:
- [x] fix regression in `media-clip-selector` - timeline hover previews were no longer showing if there is a thumbnails `<track>`
- [x] fix timeline hover previews when storyboard.vtt contains cue texts with query params (closes #33)

*Housekeeping:*

- [x] use `https` for `stream.mux` URLs in examples
- [x] cleanup some trailing whitespace in  `src/js/media-controller.js`  (we should set up a linter)
- [x] use `autohide="-1"` in `media-clip-selector` example 🎉 
- [x] on the homepage, add a link to `/examples`

mmcc commented:
I’ll dig into this more tomorrow, but is this the opportunity to move the clip selector out of the core elements dir?

heff commented on index.html:
```diff
@@ -2,8 +2,35 @@
+  </style>
 </head>
 <body>
+  <media-controller>
```
We should kill the the media-container example below this.

dylanjha replied:
looks like I messed up when I pulled in changes from `main`, will fix that

heff commented on examples/control-elements/media-clip-selector.html:
```diff
@@ -43,6 +35,7 @@
   const currentTimeEl = document.querySelector('#current-time');
   const videoEl = document.querySelector('video');
 
+  console.log('debug', videoEl.textTracks[0].mode);
```
Intending to leave this?

dylanjha replied:
nope, will drop this, good catch!

heff commented on README.md:
````diff
@@ -130,9 +130,20 @@ Use HTML to add or remove any of the controls. Then you can use CSS to style the
+Extras are not shipped with the core library, but can be imported as-needed by importing the element from the dist direcotry, for example:
+
+```js
+import 'media-chrome/dist/extras/media-clip-selector'
````
We should add the unpkg script include example here too.

heff commented on src/js/media-container.js:
```diff
@@ -25,6 +25,7 @@ template.innerHTML = `
 
       /* Default dimensions */
       width: 720px;
+      max-width: 100%;
```
What does this fix?

dylanjha replied:
This makes for a better default on smaller screens (< 720px width, like an iphone 6 that is 375px), so that it doesn't overflow horizontally

heff replied:
Oh cool, that's great. For things like that I like to add a quick comment note, otherwise the next person optimizing might strip it out not knowing why it's there.

dylanjha replied:
updated 😄

heff commented on src/js/media-controller.js:
```diff
@@ -239,7 +237,9 @@ class MediaController extends MediaContainer {
       if (volPref !== null) media.volume = volPref;
-    } catch (e) { }
+    } catch (e) {
+      console.error('Error getting volume pref', e);
```
Tracking the last volume that was set is a bonus, not a required feature, so we don't need to throw an error here. This would only fail if the browser doesn't support localStorage or its disabled, which is fine. At most I think we'd throw a `console.debug()` or similar.

dylanjha replied:
cool,  I'll use `console.debug`

heff reviewed (changes requested):
Good stuff! A few suggestions.

dylanjha commented:
@heff I addressed all your points, thanks for the review. Think we can get this change rolled out? As far as the version change goes, this will have to be 0.2 because it moves `media-clip-selector` into extras.

In the release notes we can make a note of that. I can go ahead and bump the version in this branch if that sounds good to you

heff commented:
Cool, merging in. I use `np` to push releases so no need to manually version it.

dylanjha commented:
@heff awesome, well I was going to mention this but you already got it 😄 

https://github.com/muxinc/media-chrome/commit/89cfa08e3a00a52b0d845ccb7e5f3c9aca761b53
